Library/Homebrew/utils/git.rb

```diff
@@ -16,11 +16,10 @@ def last_revision_commit_of_file(repo, file, before_commit: nil)
   def last_revision_of_file(repo, file, before_commit: nil)
     relative_file = Pathname(file).relative_path_from(repo)
 
-    commit_hash = last_revision_commit_of_file(repo, file, before_commit: before_commit)
-
+    commit_hash = last_revision_commit_of_file(repo, relative_file, before_commit: before_commit)
     out, = Open3.capture3(
       HOMEBREW_SHIMS_PATH/"scm/git", "-C", repo,
-      "show", "#{commit_hash}:#{relative_file}"
+      "show", "#{commit_hash}:#{file}"
```

Comment:
No, it should still be the relative path. Might be that `git show` also supports absolute paths, but relative is definitely cleaner.